### PR TITLE
Doc about creating custom plugin

### PR DESCRIPTION
**Pull request by estruyf**
Added the documentation to write a custom plugin for the Dev Proxy.

### CODE REVIEW

**Comment by Court72** [comment]
@estruyf - Thank you for your contribution. Someone from the content team will reach out with next steps.

**Comment by Court72** [comment]
#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

File: docs/dev/dev-proxy/how-to/create-custom-plugin.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+# Create a custom plugin
+
+In this article, you will learn how to create a custom plugin for the Dev Proxy. By creating plugins for Dev Proxy, you can extend its functionality and add custom features to fit your needs.
+
+## Prerequisites
+
+Before you start creating a custom plugin, make sure you have the following prerequisites:
+
+- [.NET Core SDK](https://dotnet.microsoft.com/download)
+- [Visual Studio Code](https://code.visualstudio.com/)
```
Comment (waldekmastykarz) [inline docs/dev/dev-proxy/how-to/create-custom-plugin.md:18]: You don't need to use VSCode. We could say we recommend it, but I wouldn't say it's a requirement.

File: docs/dev/dev-proxy/how-to/create-custom-plugin.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+      public override void Register(IPluginEvents pluginEvents,
+                                    IProxyContext context,
+                                    ISet<UrlToWatch> urlsToWatch,
+                                    IConfigurationSection? configSection = null)
+      {
+        base.Register(pluginEvents, context, urlsToWatch, configSection);
+
+        // Bind your plugin configuration
+        configSection?.Bind(_configuration);
```
Comment (waldekmastykarz) [inline docs/dev/dev-proxy/how-to/create-custom-plugin.md:92]: This is optional. Some plugins might have no configuration

**Review by waldekmastykarz — CHANGES_REQUESTED** [review]
Awesome! This is a great start for folks who are interested in building plugins. Thank you for kicking it off 👏

File: docs/dev/dev-proxy/how-to/create-custom-plugin.md (hunk 1 above)
Comment (estruyf) [reply docs/dev/dev-proxy/how-to/create-custom-plugin.md]: Removed the VSCode reference

File: docs/dev/dev-proxy/how-to/create-custom-plugin.md (hunk 2 above)
Comment (estruyf) [reply docs/dev/dev-proxy/how-to/create-custom-plugin.md:171]: I split it; the first code sample is without the configuration, and the next one with the configuration is highlighted as optional.

**Comment by estruyf** [comment]
@waldekmastykarz updated the requested changes. Let me know if anything else needs to be changed.

**Review by waldekmastykarz — APPROVED** [review]

**Comment by waldekmastykarz** [comment]
Thanks for the quick turnaround! Looks good. Let's ship it!

**Comment by estruyf** [comment]
#sign-off

**Comment by Court72** [comment]
Hi @waldekmastykarz,

This PR introduces new content that needs to be pulled into the private repo for staging, validation, and Acrolinx and to get it merged.

If you need it, the metadata/ownership info and the process for pulling content to the private repo is documented [here](https://review.docs.microsoft.com/help/contribute/contribute-how-to-manage-act-guidelines-public-pr?branch=main#pull-public-commits-to-the-private-repo-for-updates-staging-publication).

Leave a comment if you'd like the PR review team's help with moving these commits.

Thanks!

**Comment by Court72** [comment]
These changes have been merged in https://github.com/MicrosoftDocs/microsoft-cloud-pr/pull/233.

**Pull request closed without merging**